services/app-api/src/postgres/contractAndRates/prismaSharedContractRateHelpers.ts

```diff
@@ -213,6 +216,82 @@ const getParentContractID = (
 
     return parentContractID
 }
+
+const getRelatedContracts = (
+    rate: RateTableWithRelatedContractsPayload
+): RelatedContractStripped[] => {
+    const rateRevisions = [...rate.revisions].sort(
+        (revA, revB) => revA.createdAt.getTime() - revB.createdAt.getTime()
+    )
+
+    let relatedContracts: RelatedContractStripped[] = []
+    for (const rateRev of rateRevisions) {
+        if (!rateRev.submitInfo) {
+            continue
+        }
+
+        // get the latest related submissions entry
+        const latestRelatedSubmission = rateRev.relatedSubmissions.sort(
+            (a, b) => b.updatedAt.getTime() - a.updatedAt.getTime()
+        )[0]
+
+        // collect related contracts with latest consolidated statuses
+        const contracts: RelatedContractStripped[] = []
+
+        // looping throw all the submission packages of the related submissions to find the contracts for this rate.
+        latestRelatedSubmission.submissionPackages.forEach((pkg) => {
+            if (pkg.rateRevisionID === rateRev.id) {
+                let status: HealthPlanPackageStatusType = 'DRAFT'
+                let reviewStatus: ContractReviewStatusType = 'UNDER_REVIEW'
+
+                const submitInfo =
+                    pkg.contractRevision.contract.revisions[0].submitInfo
+                const unlockInfo =
+                    pkg.contractRevision.contract.revisions[0].unlockInfo
+                const latestAction =
+                    pkg.contractRevision.contract.reviewStatusActions[0]
+
+                // Find the current status of all related contracts as some could be in different status than the rate.
+                // If it has both unlock and submit info, then it was resubmitted
+                if (submitInfo && unlockInfo) {
```

Comment:
issue: We have functions capturing this logic `getContractReviewStatus`, `getRateReviewStatus`, `getContractRateStatus`.   Let's lean into or update the existing functions to make this easier to maintain otherwise we are creating another place where status could get out of sync

Reply:
🤔 I would probably make `getRateRevisionStatus`and `getContractRevisionStatus` call those both here and in the existing helpers used for other functions. Unify the logic now.

Reply:
I think it would be some work to make those functions work for both this data structure type and their existing ones. I recommend we split that work up to a follow up PR.